add Kestra flow schema

benpops89 opened the pull request:
Adds schema for [Kestra flows](https://kestra.io/docs/workflow-components/flow).

[Kestra](https://kestra.io) is an orchestration framework similar to Airflow that defines it's pipelines (flows) in YAML. This schma provides auto-complete and validation of YAML.

When validating the schema I did get the following error `Error: strict more: unknown keyword: "markdownDescription"`. It might be something specific to Kestra but not sure how it should be handled.

Solves #3978

hyperupcall commented:
> When validating the schema I did get the following error Error: strict more: unknown keyword: "markdownDescription". > It might be something specific to Kestra but not sure how it should be handled.

Schemas are validated with Ajv, which has a "strict mode" that we enable to catch common issues. More information is in [the documentation](https://github.com/SchemaStore/schemastore/blob/master/CONTRIBUTING.md#how-to-ignore-validation-errors-in-a-json-schema), but you can edit `src/schema-validation.jsonc`, and add your unknown key name to the `"options[$schemaName].unknownKeywords"` field.

benpops89 commented:
@hyperupcall - thanks for pointing me in the right direction. I have added a list of `unknownKeywords` for Kestra to that file and the checks are passing.

hyperupcall commented:
@benpops89 There appears to be one more failure in CI. The list of ignored words can be changed in `.pre-commit-config.yaml`

benpops89 commented:
@hyperupcall - thanks for your support on this. I have run codespell locally and fixed all the typos in the file and added one word `CopyIn` to the list of ignored words and the tests are now passing 🚀

hyperupcall commented:
Awesome, thank you!
